plugins/modules/vmware_drs_group.py

```diff
@@ -307,17 +301,12 @@ def __set_host_obj_list(self, host_list=None):
             host_list = self.__host_list
 
         if host_list is not None:
-
             for host in host_list:
-
                 if self.module.check_mode is False:
```

Comment:
```suggestion
                if not self.module.check_mode:
```
Since you changed `self.module.check_mode is False` to `not self.module.check_mode` in other lines, don't you want  to change it in this line also?

The rest LGTM.